wallet/core/src/storage/keydata/assoc.rs

```diff
@@ -0,0 +1,89 @@
+use crate::imports::*;
+
+#[derive(Debug, Clone, Serialize, Deserialize, BorshSerialize, BorshDeserialize)]
+#[serde(rename_all = "lowercase")]
+#[serde(tag = "type", content = "data")]
+pub enum AssocPrvKeyDataIds {
+    None,
+    Single(PrvKeyDataId),
+    Multiple(Arc<Vec<PrvKeyDataId>>),
+}
+
+impl IntoIterator for &AssocPrvKeyDataIds {
+    type Item = PrvKeyDataId;
+    type IntoIter = std::vec::IntoIter<Self::Item>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        match self {
+            AssocPrvKeyDataIds::None => Vec::new().into_iter(),
+            AssocPrvKeyDataIds::Single(id) => vec![*id].into_iter(),
+            AssocPrvKeyDataIds::Multiple(ids) => (**ids).clone().into_iter(),
+        }
+    }
+}
```

Comment:
can be done without allocations using itertool::Either.  implement Iterator over AssocPrvKeyDataIds instead

Reply:
Refactored in https://github.com/aspectron/rusty-kaspa/commit/a05ca54a275baad58df947aa3d16678e77a8ae53